src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -28,38 +147,398 @@ const ActorInit En_Door_InitVars = {
+        if (objectBankIndex != 0) {
+            Actor_MarkForDeath(&this->dyna.actor);
+            return;
+        }
+    }
+    this->requiredObjBankIndex = objectBankIndex;
+    this->dlIndex = objectInfo->dListIndex; // Set twice?
+    if (this->dyna.actor.objBankIndex == this->requiredObjBankIndex) {
+        func_80866A5C(this, globalCtx);
+    } else {
+        this->actionFunc = func_80866A5C;
+    }
+    Actor_SetHeight(&this->dyna.actor, 35.0f);
+}
 
+void EnDoor_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDoor* this = (EnDoor*)thisx;
+
+    if (this->unk_1A4 != 7) {
+        TransitionActorEntry* transitionEntry =
+            &globalCtx->doorCtx.transitionActorList[(u16)this->dyna.actor.params >> 0xA];
+        if (transitionEntry->id < 0) {
+            transitionEntry->id = -transitionEntry->id;
+        }
+    } else if (this->switchFlag == 0) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void func_80866A5C(EnDoor* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->requiredObjBankIndex)) {
+        this->dyna.actor.objBankIndex = this->requiredObjBankIndex;
+        this->actionFunc = func_80866B20;
+        this->dyna.actor.world.rot.y = 0;
+        if (this->unk_1A4 == 1) {
+            if (!Flags_GetSwitch(globalCtx, this->switchFlag)) {
+                this->unk_1A6 = 10;
+            }
+        } else if ((this->unk_1A4 == 4) &&
+                   (Actor_XZDistanceBetweenActors(&this->dyna.actor, &GET_PLAYER(globalCtx)->actor) > 120.0f)) {
+            this->actionFunc = func_8086704C;
+            this->dyna.actor.world.rot.y = -0x1800;
+        }
+    }
+}
+
+#ifdef NON_EQUIVALENT
+void func_80866B20(EnDoor* this, GlobalContext* globalCtx) {
+    static s32 D_80867BC0[4];
+
+    Player* player;
+    Vec3f playerPosRelToDoor;
+    struct_80133038_arg2 sp30;
+    s16 temp_a2;
+    s16 yawDiff;
+    s32 temp_a1_2;
+    s32 temp_t0;
+    u8 temp_a1;
+    // s16 baseTextId;
+
+    player = GET_PLAYER(globalCtx);
+    if ((func_800B84D0(&this->dyna.actor, globalCtx) != 0) && (this->dyna.actor.textId == 0x1821)) {
+        D_80867BC0[0] = 1;
+    }
+    if (this->unk_1A1 != 0) {
+        this->actionFunc = func_80867144;
+        Animation_PlayOnceSetSpeed(&this->skelAnime, sAnimations[this->animIndex],
+                                   (player->stateFlags1 & 0x8000000) ? 0.75f : 1.5f);
+        if (this->unk_1A6 != 0) {
+            gSaveContext.inventory.dungeonKeys[gSaveContext.mapIndex]--;
+            Actor_SetSwitchFlag(globalCtx, this->switchFlag);
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_CHAIN_KEY_UNLOCK);
+        }
+    } else if (this->unk_1A7 != 0) {
+        this->actionFunc = func_80866F94;
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_DOOR_OPEN);
+    } else if (func_801233E4(globalCtx) == 0) { // Player_InCsMode
```

Comment:
Since you now this function is boolean, shouldn't you treat it like that a boolean check then?
```suggestion
    } else if (!func_801233E4(globalCtx)) {
```
(And yeah, I can confirm this function is `Player_InCsMode`, because I have it decomped in the player_lib branch)